casper/src/test/resources/PoSTest.rho

```diff
@@ -193,37 +193,39 @@ match (
                     @initialRewards <- initialRewardsCh) {
                   @PoS!("pay", 100, *retCh) |
                   for ( @(true, _) <- retCh) {
-
-                    @posVault!("balance", *finalPosBalanceCh) |
-                    @user1Vault!("balance", *finalUser1BalanceCh) |
-                    @PoS!("getRewards", *finalRewardsCh) |
-                    for(@finalPosBalance <- finalPosBalanceCh;
-                        @finalUser1Balance <- finalUser1BalanceCh;
-                        @finalRewards <- finalRewardsCh) {
-                      @ListOps!("fold", finalRewards.toList(), {}, *computeDelta, *deltaRewardsCh) |
-                      contract computeDelta(@(pk, finalReward), @acc, resultCh) = {
-                        resultCh!(acc.set(pk, finalReward - initialRewards.get(pk)))
-                      } |
-                      for (@deltaRewards <- deltaRewardsCh) {
-                        @ListOps!("forall", deltaRewards, *isPositive, *allRewardsPositiveCh) |
-                        contract isPositive(@n, resultCh) = { resultCh!(n > 0)} |
-
-                        @ListOps!("fold", deltaRewards.toList(), 0, *sumDelta, *deltaSumCh) |
-                        contract sumDelta(@(_, delta), @acc, resultCh) = {
-                          resultCh!(acc + delta)
+                    @PoS!("closeBlock", *ackCh) |
+                    for (_ <- ackCh) {
+                      @posVault!("balance", *finalPosBalanceCh) |
+                      @user1Vault!("balance", *finalUser1BalanceCh) |
+                      @PoS!("getRewards", *finalRewardsCh) |
+                      for(@finalPosBalance <- finalPosBalanceCh;
+                          @finalUser1Balance <- finalUser1BalanceCh;
+                          @finalRewards <- finalRewardsCh) {
+                        @ListOps!("fold", finalRewards.toList(), {}, *computeDelta, *deltaRewardsCh) |
+                        contract computeDelta(@(pk, finalReward), @acc, resultCh) = {
+                          resultCh!(acc.set(pk, finalReward - initialRewards.get(pk)))
                         } |
+                        for (@deltaRewards <- deltaRewardsCh) {
+                          @ListOps!("forall", deltaRewards, *isPositive, *allRewardsPositiveCh) |
```

Comment:
Please change this test to assert that each validator's reward has increased by the proportion of the payment amount corresponding to its weight.

Reply:
Is it more complicated than running a `forall` on all elements of the `finalRewards` map?

Reply:
The test is too big anyway and both with yours and my proposal we test the same call paths but mine is simpler.